From: Liam Howlett <liam.howlett@oracle.com>
Subject: mmap: remove unroll from __vma_adjust()
Date: Tue, 19 Jul 2022 18:49:48 +0000

If __vma_adjust() fails, then there is no reason to undo the anon vma work
since it will be done in the caller or we have been OOMed.

Link: https://lkml.kernel.org/r/20220719184938.3359932-1-Liam.Howlett@oracle.com
Fixes: 83651ef5048d ("mm: start tracking VMAs with maple tree")
Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Suggested-by: Hugh Dickins <hughd@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mmap.c |    6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

--- a/mm/mmap.c~mm-start-tracking-vmas-with-maple-tree-fix
+++ a/mm/mmap.c
@@ -923,12 +923,8 @@ int __vma_adjust(struct vm_area_struct *
 		}
 	}
 	vma_adjust_trans_huge(orig_vma, start, end, adjust_next);
-
-	if (mas_preallocate(&mas, vma, GFP_KERNEL)) {
-		if (exporter && exporter->anon_vma)
-			unlink_anon_vmas(importer);
+	if (mas_preallocate(&mas, vma, GFP_KERNEL))
 		return -ENOMEM;
-	}
 
 	if (file) {
 		mapping = file->f_mapping;
_
